DynamicIntArray.cpp

```diff
@@ -0,0 +1,90 @@
+#include "DynamicIntArray.h"
+#include <iostream>
+
+DynamicIntArray::DynamicIntArray(): m_size(0), m_array(new int[0] {}) {};
+
+DynamicIntArray::DynamicIntArray(std::size_t size): m_size(size), m_array(new int[size] {}) {};
+
+DynamicIntArray::DynamicIntArray(const DynamicIntArray& other) {
+
+	m_size = other.m_size;
+	m_array = new int[m_size];
+	
+	for (int i = 0; i < m_size; i++) {
+		m_array[i] = other.m_array[i];
+	}
+}
+
+DynamicIntArray& DynamicIntArray::operator=(const DynamicIntArray& other) {
+
```

Comment:
self assignment check is required.
What if it's called that way?
array1 = array1;